polars/polars-core/src/frame/groupby/pivot.rs

```diff
@@ -226,71 +227,81 @@ impl DataFrame {
         // for every column where the values are aggregated
         let df_cols = (0..columns.len())
             .zip(columns_unique)
-            .flat_map(|(i, unique_vals)| {
-                let im_results = im_result
-                    .iter_mut()
-                    .map(|v| std::mem::take(&mut v[i + 1]))
+            .flat_map(|(column_index, unique_vals)| {
+                // the values that will be the new headers
+                // let unique_df = DataFrame::new_no_checks(vec![unique_vals.clone()]);
+                // let unique_set = PlHashSet::from_iter(unique_vals.iter());
+
+                // Join every row with the unique column. THis join is needed because some rows don't all values and we want to have
```

Comment:
`Oin every row with the nuique column. this Join is needed because some rows don't have all values and we want to have`